transport/cert/dca.go

```diff
@@ -43,7 +42,7 @@ const (
 // GetClientCertificateSourceAndEndpoint is a convenience function that invokes
 // getClientCertificateSource and getEndpoint sequentially and returns the client
 // cert source and endpoint as a tuple.
-func GetClientCertificateSourceAndEndpoint(settings *internal.DialSettings) (cert.Source, string, error) {
+func GetClientCertificateSourceAndEndpoint(settings *internal.DialSettings) (Source, string, error) {
```

Comment:
I don't think this package can be made public as is. The input to this function is a struct that can't be made public. I think for now a little bit of copying is the right answer. This change would add quite a bit of public surface.